tests/unit/test_charm.py

```diff
@@ -186,7 +197,65 @@ def test_on_start_not_ready_defer(
         initialise_replica_set.assert_not_called()
 
     @patch_network_get(private_address="1.1.1.1")
-    @patch("mongod_helpers.MongoDB.is_ready")
+    @patch("mongod_helpers.MongoDB.is_replica_set")
+    @patch("mongod_helpers.MongoDB.is_replica_ready")
+    @patch("mongod_helpers.MongoDB.is_mongod_ready")
+    @patch("charm.MongodbOperatorCharm._open_port_tcp")
+    @patch("charm.MongodbOperatorCharm._initialise_replica_set")
+    @patch("charm.service_running")
+    def test_on_start_replica_not_ready_waits(
+        self,
+        service_running,
+        initialise_replica_set,
+        _open_port_tcp,
+        is_mongod_ready,
+        is_replica_ready,
+        is_replica_set,
+    ):
+        """Tests that if a replica is not ready that the charm goes into waiting."""
+        self.harness.set_leader(True)
+        is_replica_set.return_value = True
+        service_running.return_value = True
+        is_mongod_ready.return_value = True
+        is_replica_ready.return_value = False
```

Comment:
When you're patching a method, `return_value` can be set in the `@patch` statements, making these lines unnecessary for simple variables like booleans.

Reply:
Great idea, since this is something that is seen in other areas of the unit tests I'll save this for my future PR where I clean up my unit tests.